core/src/components/header/header.utils.ts

```diff
@@ -0,0 +1,133 @@
+import { readTask, writeTask } from '@stencil/core';
+
+const TRANSITION = 'all 0.2s ease-in-out';
+
+export const cloneElement = (tagName: string) => {
+  const getCachedEl = document.querySelector(`${tagName}.ion-cloned-element`);
+  if (getCachedEl !== null) { return getCachedEl; }
+
+  const clonedEl = document.createElement(tagName);
+  clonedEl.classList.add('ion-cloned-element');
+  clonedEl.style.setProperty('display', 'none');
+  document.body.appendChild(clonedEl);
+
+  return clonedEl;
+};
+
+export const createHeaderIndex = (headerEl: any): any | undefined => {
+  if (!headerEl) { return; }
```

Comment:
we have some mixed styles of inline `if() + {block}` we should always do the same way. I am open for discussion, but once decided always follow the same pattern.

And add a tslint rule to ensure it.